swap_meet/vendor.py

```diff
@@ -1,2 +1,74 @@
+from .item import Item
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory:
+            self.inventory = inventory
+        else:
+            self.inventory = []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category_name):
+        res = []
+        for item in self.inventory:
+            if item.category == category_name:
+                res.append(item)
+        return res
+
+    def swap_items(self, another_vendor, my_item, their_item):
+        
+        if my_item in self.inventory and their_item in another_vendor.inventory:
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+            another_vendor.inventory.append(my_item)
+            another_vendor.inventory.remove(their_item)
+            return True
+        return False
```

Comment:
Notice how the primary logic of this function is indented? By using a guard clause we can put this more into focus. 

```python
    def swap_items(self, another_vendor, my_item, their_item):
        if my_item not in self.inventory or their_item not in another_vendor.inventory:
            return False
        self.inventory.remove(my_item)
        self.inventory.append(their_item)
        another_vendor.inventory.append(my_item)
        another_vendor.inventory.remove(their_item)
        return True
```

Reply:
We can also take advantage of the `add` and `remove` methods made for this class:

```python
    def swap_items(self, another_vendor, my_item, their_item):
        if my_item not in self.inventory or their_item not in another_vendor.inventory:
            return False
        self.remove(my_item)
        self.add(their_item)
        another_vendor.add(my_item)
        another_vendor.remove(their_item)
        return True
```

Reply:
Notice how when we changed the conditional to use `not`, we also had to change the operator from `and` to `or`.